### PR TITLE
Topic Proxy: update proxy topic's perUser from pres acs changes.

### PR DESCRIPTION
**Pull request by aforge**

### CODE REVIEW

File: server/store/types/types.go [hunk 1]
```diff
@@ -654,6 +664,75 @@ func (o AccessMode) Delta(n AccessMode) string {
...
+			default:
+				break
+			}
+		} else {
+			return err
+		}
+	}
+	*m = m0
+	return nil
+}
```
Comment (or-else) [inline server/store/types/types.go:734]: This seems a bit too complex. I would rewrite it like this: https://play.golang.org/p/Y9ZFU-Dwn5G

File: server/store/types/types.go [hunk 2]
```diff
@@ -535,9 +535,8 @@ func (m AccessMode) MarshalText() ([]byte, error) {
 	return res, nil
 }
 
-// UnmarshalText parses access mode string as byte slice.
-// Does not change the mode if the string is empty or invalid.
-func (m *AccessMode) UnmarshalText(b []byte) error {
+// DecodeAcs parses AccessMode from a byte array.
+func DecodeAcs(b []byte) (AccessMode, error) {
```
[the remark this answers is not in the document]
Comment (aforge) [reply server/store/types/types.go]: Fixed.

File: server/store/types/types.go [hunk 3]
```diff
@@ -654,6 +664,75 @@ func (o AccessMode) Delta(n AccessMode) string {
...
 
+// Applies the acs delta to AccessMode.
+// Delta is in the same format as generated by AccessMode.Delta.
+// E.g. JPRA.ApplyDelta(-PR+W) -> JWA.
+func (m *AccessMode) ApplyDelta(delta string) error {
+	if delta == "N" {
+		// No updates.
+		return nil
+	}
+	if delta == "" || (delta[0] != '+' && delta[0] != '-') {
```
[the remark this answers is not in the document]
Comment (aforge) [reply server/store/types/types.go]: Sure.

File: server/store/types/types.go (hunk 1 above)
Comment (aforge) [reply server/store/types/types.go:706]: Cool. Changed.

File: server/store/types/types.go [hunk 4]
```diff
@@ -654,6 +664,75 @@ func (o AccessMode) Delta(n AccessMode) string {
...
+		return errors.New("AccessMode: invalid acs delta string '" + delta + "'")
+	}
+	var actions, acsDeltas []string
+	acceptAction := true
+	j := -1
+	for i := 0; i < len(delta); i++ {
+		switch delta[i] {
+		case '+':
+			fallthrough
```
[the remark this answers is not in the document]
Comment (aforge) [reply server/store/types/types.go]: Done.

**Pull request merged**